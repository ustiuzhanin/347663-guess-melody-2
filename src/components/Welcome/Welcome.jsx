import React from 'react';
import PropTypes from 'prop-types';

export default function Welcome(props) {
  const {time, mistakes, onClick} = props;

  return (
    <section className='welcome'>
      <div className='welcome__logo'>
        <img
          src='img/melody-logo.png'
          alt='Угадай мелодию'
          width='186'
          height='83'
        />
      </div>
      <button className='welcome__button' onClick={onClick}>
        <span className='visually-hidden'>Начать игру</span>
      </button>
      <h2 className='welcome__rules-title'>Правила игры</h2>
      <p className='welcome__text'>Правила просты:</p>
      <ul className='welcome__rules-list'>
        <li>За {time} минут нужно ответить на все вопросы.</li>
        <li>Можно допустить {mistakes} ошибки.</li>
      </ul>
      <p className='welcome__text'>Удачи!</p>
    </section>
  );
}

Welcome.defaultProps = {
  time: 5,
  mistakes: 3,
  onClick: () => {
    // TODO: prints an error if onClick wasn't passed
  }
};

Welcome.propTypes = {
  time: PropTypes.number.isRequired,
  mistakes: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
};
