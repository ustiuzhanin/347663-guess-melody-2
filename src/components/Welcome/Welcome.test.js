import React from 'react';
import Welcome from './Welcome.jsx';
import renderer from 'react-test-renderer';

test(`Welcome's snapshot`, () => {
  const tree = renderer.create(<Welcome time={0} mistakes={0} />).toJSON();

  expect(tree).toMatchSnapshot();
});
