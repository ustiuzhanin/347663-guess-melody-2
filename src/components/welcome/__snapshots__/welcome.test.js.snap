// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome's snapshot 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[MockFunction]}
  >
    <span
      className="visually-hidden"
    >
      Play
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Game Rules
  </h2>
  <p
    className="welcome__text"
  >
    The rules are simple:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      You have to answer all the questions in 
      0
       minutes
    </li>
    <li>
      You are allowed to make 
      0
       mistakes.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Good Luck!
  </p>
</section>
`;
